test/integration/users.test.js

```diff
@@ -193,6 +204,123 @@ describe("Users", function () {
           return done();
         });
     });
+
+    it("Should return next and prev links", function (done) {
+      chai
+        .request(app)
+        .get("/users?size=2")
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Users returned successfully!");
+          expect(res.body).to.have.property("links");
+          expect(res.body.links).to.have.property("next");
+          expect(res.body.links).to.have.property("prev");
+
+          return done();
+        });
+    });
+
+    it("Should return 400 when both next and prev passed as query param", function (done) {
+      chai
+        .request(app)
+        .get(`/users?next=${userId}&prev=${userId}&size=2`)
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+
+          return done();
+        });
+    });
+
+    it("Should return 400 when both next and page passed as query param", function (done) {
+      chai
+        .request(app)
+        .get(`/users?next=${userId}&page=1&size=2`)
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+
+          return done();
+        });
+    });
+
+    it("Should return 400 when both page and prev passed as query param", function (done) {
+      chai
+        .request(app)
+        .get(`/users?page=1&prev=${userId}&size=2`)
+        .set("cookie", `${cookieName}=${jwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+
+          return done();
+        });
+    });
+
+    it("Should include all query params in the response links that are passed by the request", function (done) {
```

Comment:
this test mentions `Should include all query params` while the next test mentions `Should not have page query param`. Aren't these 2 contradictory ?

Reply:
yes will see if i can name it better

Reply:
done